src/internet_identity/src/storage.rs

```diff
@@ -145,7 +146,7 @@ pub const MAX_ENTRIES: u64 = (MAX_MANAGED_WASM_PAGES - BUCKET_SIZE_IN_PAGES as u
 
 pub type Salt = [u8; 32];
 
-type ArchiveBufferMemory<M> = RestrictedMemory<VirtualMemory<RestrictedMemory<M>>>;
+type SingleBucketMemory<M> = RestrictedMemory<VirtualMemory<RestrictedMemory<M>>>;
```

Comment:
It would be cleaner to wrap the underlying memory such that:
* the memory is guaranteed to be limited to a single bucket
* the memory cannot be written to

However, I'll do this in a separate PR.